internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -374,89 +375,116 @@ func (cf CloudFormation) executeAndRenderChangeSet(in *executeAndRenderChangeSet
 	g, ctx := errgroup.WithContext(context.Background())
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
+	prevStackEventsErased := make(chan bool)
 	g.Go(func() error {
 		defer cancel()
-		if err := cf.renderChangeSet(ctx, changeSetID, in); err != nil {
+		nl, err := cf.renderChangeSet(ctx, changeSetID, in)
+		if err != nil {
 			if !errors.Is(err, context.Canceled) {
 				return err
 			}
+			// Erase previous stack events only if context is canceled
+			// by waitForSignalAndHandleInterrupt().
+			cursor.EraseLinesAbove(cf.console, nl)
+			close(prevStackEventsErased)
```

Comment:
hmm I wonder if it is possible at all that:
1. We catched a signal, and the other go routine hangs at `<-in.prevStackEventsErased`
2. `prevStackEventsErased` is not closed
_If_ this is ever possible to happen, it _would_ result in a hanging program it seems

Reply:
I addressed this concern of hanging program in this [commit](https://github.com/aws/copilot-cli/pull/5224/commits/0c93183c8a1eeb2085b526d18d670fabd0ebba3f), I modified the logic to close the Channel always. This will ensure `waitForSignalAndHandleInterrupt`   go routine  will always completion it's execution and does not hang.